plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/PipelineStore.java

```diff
@@ -108,12 +133,14 @@ public void run() {
             try {
                 while (running) {
                     try {
-                        Thread.sleep(3000);
-                        updatePipelines();
+                        Thread.sleep(pipelineUpdateInterval.millis());
```

Comment:
this whole updater should be based on scheduled executor, where every second you'll run "update" task and when you're done you'll schedule another update task for a second later